src/lib/Sockets/Socket.ts

```diff
@@ -5,7 +5,7 @@ export default class Socket {
 
   constructor(
       public name: string,
-      public valueTypeName: string,
+      public valueTypeName: 'flow' | 'string' | 'number' | 'boolean',
```

Comment:
FYI: I recently added ValueTypeRegistry that allows you to register new types.  And when you register new types you have to provide a string name as well as both a "toString" and a "parse" function.  This was to allow for adding new types and not being fixed.  The read/write to JSON function now uses toString and Parse to save/load the values from strings in the JSON file.  This means that this probably shouldn't be fixed to just these types as we can now expand past them.

The idea is to allow others to use this library with their own types, rather than just being fixed to these four.

Reply:
The ValueTypeRegistry is here:

https://github.com/bhouston/behave-graph/blob/main/src/lib/Values/ValueTypeRegistry.ts

Here is where I use the valueTypeRegistry to parse the JSON values:

https://github.com/bhouston/behave-graph/blob/main/src/lib/Graphs/IO/readGraphFromJSON.ts#L47

Reply:
Ok that makes sense. Let me go back and have another go. There may be a nicer way to do it now with generics that would could use. I'd hate to just leave it as string as I rely on it in the UI to determine what type of control to show, and just leaving it as string feels potentially fragile

Reply:
i've dug into the new ValueTypeRegistry system now and got my head around it and I understand why these needs to be strings (at the JSON stage at least), so i've reverted the highlighted changes.